### DISCUSSION TITLE
Global styles and scripts

### DISCUSSION THREAD

**Discussion by ale1981** [Laravel-Backpack/community-forum#744] [Q&A (Help)] [answered]
Is there any reason CDNs that I add to the global sections in ui.php wouldn't load on every page load?
Also, how would I get these to work with Basset?

If I manually add them to a view, they work fine.

I am using the default Tabler theme.

**Comment by jcastroa87** [comment]
Hello @ale1981 , can you please detail the problem, in which sections it loads and in which it does not?

Did you clean the cache after making the change?
```
php artisan optimize:clear
```

To handle it with basset, you can check instructions [here](https://github.com/Laravel-Backpack/basset)

If you have any other problem, please tell us and send the output of to have the full panorama
```
php artisan backpack:version
```

Thanks.

**Comment by ale1981** [comment]
Hi @jorgetwgroup thanks for your response.

As an example, I have some fa icons in the menu and by adding the font awesome CSS file to the Global Styles section of the ui.php file, it doesn't work, the icons don't appear.

If I add the CSS file manually using ```@basset(https://cdnjs.cloudflare.com/ajax/libs/font-awesome/6.4.2/css/all.min.css')``` to the ```menu_items.blade.php``` file, then the icons work as expected.

This doesn't work...
```
    // -------------
    // GLOBAL STYLES
    // -------------

    // CSS files that are loaded in all pages, using Laravel's asset() helper
    'styles' => [
        // 'styles/example.css',
        // 'https://some-cdn.com/example.css',
        'https://cdnjs.cloudflare.com/ajax/libs/font-awesome/6.4.2/css/all.min.css',
    ],
```

I have read the basset help, but doesn't explain how I would utilise basset from inside the ui.php file to add packages globally?
If I try the following, I get a 500 facade root not set error...

```
    // -------------
    // GLOBAL STYLES
    // -------------

    // CSS files that are loaded in all pages, using Laravel's asset() helper
    'styles' => [
        // 'styles/example.css',
        // 'https://some-cdn.com/example.css',
        basset('https://cdnjs.cloudflare.com/ajax/libs/font-awesome/6.4.2/css/all.min.css'),
    ],
```

```
### PHP VERSION:
PHP 8.1.2-1ubuntu2.14 (cli) (built: Aug 18 2023 11:41:11) (NTS)
Copyright (c) The PHP Group
Zend Engine v4.1.2, Copyright (c) Zend Technologies
    with Zend OPcache v8.1.2-1ubuntu2.14, Copyright (c), by Zend Technologies

### LARAVEL VERSION:
10.32.1.0

### BACKPACK PACKAGE VERSIONS:
backpack/basset: 1.2.1
backpack/crud: 6.3.0
backpack/generators: v4.0.2
backpack/pro: 2.0.18
backpack/theme-tabler: 1.1.1
```

**Reply by pxpm** [reply, marked as answer]
Hey @ale1981 the reason why backpack is not picking your `styles` definition on the `ui.php` config file, is because tabler theme overwrites that `styles` key, so you need to configure it on theme config file, that takes precedence over the ui file. 

If you haven't published tabler configuration yet, you can do it with: `php artisan vendor:publish --tag="theme-tabler-config"`that will publish to your application the tabler config file, where you can find : https://github.com/Laravel-Backpack/theme-tabler/blob/1afe83382194d776717804ca9ea4a1e82e398ccc/config/theme-tabler.php#L41 to add your styles.

No need to overwrite files, I think @jorgetwgroup may have misunderstood what you wanted to achieve here. 

Cheers

**Reply by ale1981** [reply]
Hi @pxpm 

I already had the tabler config file published, how would I utilise ```basset``` from within that file or does it use ```basset``` as standard for anything custom added to the ```styles``` and ```scripts``` keys?

Thanks

**Reply by pxpm** [reply]
Hey @ale1981 it will run the urls/paths you put there through basset yes: https://github.com/Laravel-Backpack/CRUD/blob/4c7030524540d34ca32db367a48ae03d34c14505/src/resources/views/ui/inc/styles.blade.php#L11

Cheers

**Reply by ale1981** [reply]
Hi @pxpm, perfect, I added a scripts key as it didn't exist in the theme-tabler config file and both styles and scripts seem to work great.

Is there any documentation on this that I missed?

**Reply by pxpm** [reply]
Very glad it's working for you 🙏 

See if [this helps]( https://backpackforlaravel.com/docs/6.x/base-themes#theme-configs-1) in understanding the themes and how the link/overwrite of the core components works.

Cheers

**Comment by jcastroa87** [comment]
Hello @ale1981 

I find this way we load font awesome on Backpack using basset https://github.com/Laravel-Backpack/CRUD/blob/7e10acf5f60156964ebd25e8481f317826ea86e0/src/resources/views/ui/inc/styles.blade.php#L4

You can try to override and load your own package as you need.

Cheers.

**Reply by ale1981** [reply]
Hi @jorgetwgroup 

Thanks for that, I was hoping that I wouldn't have to override the default files but will give that a go 👍

**Comment by jcastroa87** [comment]
Great @ale1981 , i will close the issue, in any case, please feel free to re-open or create a new one if needed.

Thanks.
